sled-agent/src/bootstrap/params.rs

```diff
@@ -96,6 +183,24 @@ pub struct RackInitializeRequest {
     pub allowed_source_ips: AllowedSourceIps,
 }
 
+impl RackInitializeRequest {
+    pub fn from_toml_with_fallback(
+        data: &str,
+    ) -> Result<RackInitializeRequest> {
+        let v2_err = match toml::from_str::<RackInitializeRequest>(&data) {
+            Ok(req) => return Ok(req),
+            Err(e) => e,
+        };
+        if let Ok(v1) =
+            toml::from_str::<back_compat::RackInitializeRequestV1>(&data)
+        {
+            return Ok(v1.into());
+        }
+
+        Err(v2_err.into())
```

Comment:
I'm assuming the intent is: If we can't parse this struct as any known format, return the parsing error for the most-recent version?

Reply:
Exactly.  I'll add a comment to that effect.